osf_tests/test_guid.py

```diff
@@ -452,3 +457,77 @@ def test_resolve_guid_download_errors(self):
 
         res = self.app.get(pp.url + 'download', auth=non_contrib.auth)
         assert res.status_code == 410
+
+@pytest.fixture()
+def creator():
+    return AuthUserFactory()
+
+@pytest.fixture()
+def auth(creator):
+    return Auth(user=creator)
+
+@pytest.fixture()
+def preprint_provider():
+    return PreprintProviderFactory()
+
+@pytest.mark.django_db
+class TestGuidVersionsThrough:
+    def test_creation_versioned_guid(self, creator, preprint_provider):
+        preprint = Preprint.create(
+            creator=creator,
+            provider=preprint_provider,
+            title='Preprint',
+            description='Abstract'
+        )
+
+        version_entry = preprint.versioned_guids.first()
+        assert preprint.creator == creator
+        assert preprint.provider == preprint_provider
+        assert preprint.title == 'Preprint'
+        assert preprint.description == 'Abstract'
+
+        assert version_entry.guid == preprint.guids.first()
+        assert version_entry.referent == preprint
+        assert version_entry.content_type.model == 'preprint'
+        assert version_entry.object_id == preprint.pk
+        assert version_entry.version == 1
```

Comment:
We need to test both the `Guid` and `GuidVersionsThrough`. I think you have most of them but let's be thorough.

* Check all fields of the `Guid` object of which this preprint is a referent

* Check all fields of the `GuidVersionsThrough` object of which this preprint is a referent

In addition, make sure:

* `preprint.guids` only has one element, which is true no matter how many versions there are.

* `preprint.versioned_guids` only has one element (due to it there is only one version)

Reply:
Added new checks